### ISSUE TITLE
About the multi-process support

### ISSUE THREAD

**Issue by elct9620** [yabeda-rb/yabeda#2]
Hi, there

I test `yabeda-promtheus` with Passenger, but it seems not working. But the local machine is working well with puma.

After I read the source code, I guess it has some problem when running with a multi-process web server.

In this case, I think to add an option for the user to choose use `concurrent-ruby` in memory or cache the metrics in another server, ex. redis.

> I read the source code about the ActionCable one year ago, it also uses Redis (or PostgreSQL) to shared some connection between process/server. So I think add Redis support is a good idea.

**Comment by Envek** [comment]
Yes, in case of yabeda-prometheus it wraps [original `prometheus-client`](https://github.com/prometheus/client_ruby/) which have [open issue about multi-process support](https://github.com/prometheus/client_ruby/issues/9). 

GitLab forked prometheus-client to [`prometheus-client-mmap`](https://gitlab.com/gitlab-org/prometheus-client-mmap/) which does support multi-process servers via memory mapped files. I believe that they are totally compatible so that it possible to switch yabeda-prometheus over these two. I will investigate that.

We're not faced with this problem because if we need to run multiple Passengers (or Pumas) we do not use their cluster possibilities to launch multiple processes at the same host but run multiple containers with single process in each and let Prometheus to scrape each container individually.

Another option you can consider for now is [`prometheus_exporter`](https://github.com/discourse/prometheus_exporter) gem by Discourse. It much more mature by now, support multi-process servers, have a lot of metrics but very opinionated for us and supports only Prometheus.
